File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    idx = 0
    day = 1
    cnt = 0
    # for i in range(len(speeds)):
    while idx < len(speeds):
        progress = progresses[idx]
        if progress + day * speeds[idx] >= 100:
            cnt += 1
            idx += 1
            continue

        if cnt == 0:
            day += 1
            continue

        answer.append(cnt)
        cnt = 0

    answer.append(cnt)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int day = (100 - progresses[0]) / speeds[0] + ((100 - progresses[0]) % speeds[0] != 0 ? 1 : 0);
    int cnt = 0;
    for(int i=0;i<progresses.size();i++){
        int now = progresses[i] + speeds[i] * day;
        if(now >=100){
            cnt++;
            continue;
        }
        else{
            answer.push_back(cnt);
            cnt=1;
            day += (100-now) / speeds[i];
            if((100-now) % speeds[i] != 0) day++;
        }
    }
    answer.push_back(cnt);
    return answer;
}